implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/PrometheusHistogram.java

```diff
@@ -107,16 +113,20 @@ private void updateExemplar(double value, @Nullable TimeUnit sourceUnit, @Nullab
             int index) {
         double bucketFrom = (index == 0) ? Double.NEGATIVE_INFINITY : buckets[index - 1];
         double bucketTo = buckets[index];
-        Exemplar prev;
-        Exemplar next;
+        Exemplar previusBucketExemplar;
+        Exemplar previousLastExemplar;
+        Exemplar nextExemplar;
 
         double exemplarValue = (sourceUnit != null && destinationUnit != null)
                 ? TimeUtils.convert(value, sourceUnit, destinationUnit) : value;
         do {
-            prev = exemplars.get(index);
-            next = exemplarSampler.sample(exemplarValue, bucketFrom, bucketTo, prev);
+            previusBucketExemplar = exemplars.get(index);
+            previousLastExemplar = lastExemplar.get();
+            nextExemplar = exemplarSampler.sample(exemplarValue, bucketFrom, bucketTo, previusBucketExemplar);
```

Comment:
Please notice that here and both in `PrometheusDistributionSummary`/`PrometheusTimer` the value of the last exemplar is the recorded value. I think this is expected, but the value for `_count` should not be the value that was recorded but (in these cases) `1.0`.
Because of this the value is set in the registry (see my comment on it).